src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -414,40 +746,45 @@ class SegmentCleaner : public SegmentProvider {
   /// head of journal
   journal_seq_t journal_head;
 
+  device_id_t journal_device_id;
+
   ExtentCallbackInterface *ecb = nullptr;
 
   /// populated if there is an IO blocked on hard limits
   std::optional<seastar::promise<>> blocked_io_wake;
 
+  std::vector<device_id_t> effective_devices;
+
 public:
   SegmentCleaner(
     config_t config,
-    ScannerRef&& scanner,
+    ExtentReaderRef&& scanner,
     bool detailed = false);
 
-  void mount(SegmentManager &sm) {
+  void mount(SegmentManager &psm, std::vector<SegmentManager*>& sms) {
+    crimson::get_logger(ceph_subsys_seastore).debug(
+      "SegmentCleaner::mount: {} segment managers", sms.size());
     init_complete = false;
     used_bytes = 0;
     journal_tail_target = journal_seq_t{};
     journal_tail_committed = journal_seq_t{};
     journal_head = journal_seq_t{};
+    journal_device_id = psm.get_device_id();
```

Comment:
Why not just pass in this device id as journal_device_id?

Reply:
there are no device id field in SegmentCleaner

Reply:
SegmentCleaner::journal_device_id.  SegmentCleaner::mount() here accepts a SegmentManager reference, psm, which is only used to obtain an id to populate journal_device_id here.  Just pass in the device_id_t rather than the whole SegmentManager.

Reply:
Ah, got it.